apm-sniffer/apm-sdk-plugin/spring-plugins/spring-webflux-5.x-webclient-plugin/src/main/java/org/apache/skywalking/apm/plugin/spring/webflux/v5/webclient/WebFluxWebClientInterceptor.java

```diff
@@ -63,12 +65,12 @@ public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allAr
         if (request instanceof EnhancedInstance) {
             ((EnhancedInstance) request).setSkyWalkingDynamicField(contextCarrier);
         }
-        
+
         //user async interface
         span.prepareForAsync();
         ContextManager.stopSpan();
 
-        objInst.setSkyWalkingDynamicField(span);
+        spanLocal.set(span);
```

Comment:
Check plugin API, we have MIC to propagate this reference to after stage, and 100% safe, also have better performance.

Reply:
Please check the new commit.